src/azure/azureController.ts

```diff
@@ -117,48 +117,65 @@ export class AzureController {
 		await this.prompter.promptSingle(tenantQuestion, true);
 	}
 
-	public async getTokens(profile: ConnectionProfile, accountStore: AccountStore, settings: AADResource): Promise<ConnectionProfile> {
+	public async addAccount(accountStore: AccountStore): Promise<IAccount> {
 		let account: IAccount;
 		let config = vscode.workspace.getConfiguration('mssql').get('azureActiveDirectory');
 		if (config === utils.azureAuthTypeToString(AzureAuthType.AuthCodeGrant)) {
 			let azureCodeGrant = await this.createAuthCodeGrant();
 			account = await azureCodeGrant.startLogin();
 			await accountStore.addAccount(account);
-			if (!profile.tenantId) {
-				await this.promptForTenantChoice(account, profile);
-			}
-			let tid = profile.tenantId ? profile.tenantId : azureCodeGrant.getHomeTenant(account).id;
-			const token = await azureCodeGrant.getAccountSecurityToken(
-				account, tid, settings
-			);
-			if (!token) {
-				let errorMessage = LocalizedConstants.msgGetTokenFail;
-				this._vscodeWrapper.showErrorMessage(errorMessage);
-			}
-			profile.azureAccountToken = token.token;
-			profile.expiresOn = token.expiresOn;
-			profile.email = account.displayInfo.email;
-			profile.accountId = account.key.id;
 		} else if (config === utils.azureAuthTypeToString(AzureAuthType.DeviceCode)) {
```

Comment:
@cssuh before my change when config was AzureAuthType.DeviceCode, the getToken method didn't set the account and token in the profile. Should I keep it that way or was it a bug?

Reply:
hmm, that seems like a bug.  I think it should be set.

Reply:
ok I keep the change then